common/src/main/java/com/datastax/oss/common/sink/config/ContactPointsValidator.java

```diff
@@ -40,8 +40,8 @@ public static void validateContactPoints(List<String> contactPoints) {
     if (!invalid.isEmpty()) {
       throw new ConfigException(
           String.format(
-              "Incorrect %s: %s",
-              CassandraSinkConfig.CONTACT_POINTS_OPT, String.join(",", invalid)));
+              "Incorrect %s: %s : The %s should only include DNS names or IP addresses (without the CQL port)." ,
```

Comment:
What about: "The %s **configuration entry** should only include..."

Reply:
@yabinmeng, I agree with you but the driver team was involved in this code, maybe there was a good reason behind that choice ? @adutra @eolivelli, should we change that ?

Reply:
I wouldn't say it's "common practice" to have both the host and port in each contact point string.

Different port numbers per node is a feature only available in Cassandra 4.0, see [CASSANDRA-7544](https://issues.apache.org/jira/browse/CASSANDRA-7544). And still, most deployments out there still use the same port since it's easier to manage in containerized environments.

This is the reason why the old Kafka Sink Connector had a separate option for the port: it simply assumes that we are using the same port for all contact points. This is also the case for most drivers, and old versions of DSBulk. Only the new Java driver 4.x (and DSbulk 1.4+) changed that and now requires the port to be passed for each contact point, precisely because of CASSANDRA-7544.

My opinion is that indeed we could change that in the future, but my gut feeling is that it is not super urgent either.